DNN Platform/Library/Security/Membership/AspNetMembershipProvider.cs

```diff
@@ -1841,7 +1839,11 @@ public override void UpdateUsersOnline(Hashtable userList)
                 {
                     //Clear the user object
                     user = null;
-                }                              
+
+                    //Clear cache for user so that locked out & other status could be updated
+                    DataCache.ClearUserCache(portalId, username);
+                    DataCache.ClearCache(GetCacheKey(username));
```

Comment:
Also, why is cache cleared inside the `if (!bValid)` block?

Reply:
Its mean Username or Email exist but it has failure because of password mismatched or any other reason, clearing these cache will make framework to retrieve fresh info regarding lockedout or other.